<a href="https://colab.research.google.com/github/elleniayele/Prediction-of-Product-Sales1/blob/main/Average_Height_per_individual_olympian_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [9]:
import pandas as pd
import numpy as np


In [10]:
df = pd.read_csv("/content/athleteEventsNoPersonal.csv")

# New Section

In [11]:
df.head()

ID   Age  Height         Team  NOC        Games  Year  Season         City  \
0   5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter      Calgary   
1   5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter      Calgary   
2   5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter  Albertville   
3   5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter  Albertville   
4   5  27.0   185.0  Netherlands  NED  1994 Winter  1994  Winter  Lillehammer   

           Sport                               Event Medal  
0  Speed Skating    Speed Skating Women's 500 metres   NaN  
1  Speed Skating  Speed Skating Women's 1,000 metres   NaN  
2  Speed Skating    Speed Skating Women's 500 metres   NaN  
3  Speed Skating  Speed Skating Women's 1,000 metres   NaN  
4  Speed Skating    Speed Skating Women's 500 metres   NaN

how many rows?
# New Section
how many columns?

In [12]:
df.shape

(40616, 12)

3.Which columns have missing values? (Why do you think there are there so many null values in the Medals column?)




In [13]:
df.isna().sum()

ID            0
Age        1473
Height     9001
Team          0
NOC           0
Games         0
Year          0
Season        0
City          0
Sport         0
Event         0
Medal     34699
dtype: int64

column Age, Height and Medal have missing values.
the reason why this athlete has missing values in the medal column is because he hasnt won and earned a medal in the particular olympics.

4.How many entries correspond to the city of London?

In [14]:
df1= df[df['City']=='London']
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3370 entries, 36 to 40609
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      3370 non-null   int64  
 1   Age     3067 non-null   float64
 2   Height  2139 non-null   float64
 3   Team    3370 non-null   object 
 4   NOC     3370 non-null   object 
 5   Games   3370 non-null   object 
 6   Year    3370 non-null   int64  
 7   Season  3370 non-null   object 
 8   City    3370 non-null   object 
 9   Sport   3370 non-null   object 
 10  Event   3370 non-null   object 
 11  Medal   568 non-null    object 
dtypes: float64(2), int64(2), object(8)
memory usage: 342.3+ KB


there are 3370 entries

5.What age is the youngest athlete in our sample data? Hint: use min()

In [15]:
df['Age'].min()

11.0

Create filters to find out:
How many athletes who participated in the Sport "Tug-Of-War" were from Team "Sweden" in our sample dataset?

In [16]:
df2=df.loc[(df['Sport']=="Tug-Of-War")&(df['Team']=='Sweden')]
df2

ID   Age  Height    Team  NOC        Games  Year  Season    City  \
10023  34205  22.0     NaN  Sweden  SWE  1908 Summer  1908  Summer  London   
12681  43105  23.0     NaN  Sweden  SWE  1906 Summer  1906  Summer  Athina   
13015  44087  29.0   182.0  Sweden  SWE  1906 Summer  1906  Summer  Athina   
16250  55169  23.0     NaN  Sweden  SWE  1908 Summer  1908  Summer  London   
19048  64266  21.0     NaN  Sweden  SWE  1908 Summer  1908  Summer  London   
20550  68647  26.0   190.0  Sweden  SWE  1906 Summer  1906  Summer  Athina   

            Sport                        Event   Medal  
10023  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN  
12681  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze  
13015  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze  
16250  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN  
19048  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN  
20550  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze

so there are 6 athelets who participated in Team sweden with in the sport Tug-Of-War

How many Gold medals were awarded to athletes who were over 40 years old?

In [17]:
df3=df[(df['Medal']=='Gold')&(df['Age']> 40)]
df3

ID   Age  Height                Team  NOC        Games  Year  \
740      2735  41.0   168.0              Russia  RUS  2000 Summer  2000   
1103     4029  46.0     NaN           Argentina  ARG  1936 Summer  1936   
1421     4993  41.0   180.0             Mosk II  NOR  1920 Summer  1920   
3040    10769  49.0   180.0          White Lady  DEN  1964 Summer  1964   
3270    11617  44.0     NaN             Heroine  GBR  1908 Summer  1908   
...       ...   ...     ...                 ...  ...          ...   ...   
38916  130271  43.0     NaN            Heira II  NOR  1920 Summer  1920   
38952  130381  42.0   167.0       United States  USA  1972 Summer  1972   
39193  130936  56.0     NaN       United States  USA  1908 Summer  1908   
39197  130936  60.0     NaN       United States  USA  1912 Summer  1912   
39365  131533  58.0     NaN  Cincinnati Archers  USA  1904 Summer  1904   

       Season       City             Sport  \
740    Summer     Sydney          Shooting   
1103   Summer     Berlin              Polo   
1421   Summer  Antwerpen           Sailing   
3040   Summer      Tokyo           Sailing   
3270   Summer     London           Sailing   
...       ...        ...               ...   
38916  Summer  Antwerpen           Sailing   
38952  Summer     Munich           Archery   
39193  Summer     London          Shooting   
39197  Summer  Stockholm  Art Competitions   
39365  Summer  St. Louis           Archery   

                                             Event Medal  
740    Shooting Men's Rapid-Fire Pistol, 25 metres  Gold  
1103                               Polo Men's Polo  Gold  
1421                       Sailing Mixed 10 metres  Gold  
3040           Sailing Mixed Three Person Keelboat  Gold  
3270                        Sailing Mixed 7 metres  Gold  
...                                            ...   ...  
38916                      Sailing Mixed 12 metres  Gold  
38952                   Archery Women's Individual  Gold  
39193   Shooting Men's Running Target, Double Shot  Gold  
39197           Art Competitions Mixed Sculpturing  Gold  
39365                   Archery Women's Team Round  Gold  

[66 rows x 12 columns]

so there are 66 athletes whom won Gold medal with age >40.

3.1 Determine the average height if all rows are included.

In [18]:
df['Height'].mean()

175.3811165585956

All rows are included but grouped by Event.

In [19]:
df.groupby('Event').mean()['Height']

<ipython-input-19-9bf18d3cc583>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby('Event').mean()['Height']


Event
Alpine Skiing Men's Combined                      178.547170
Alpine Skiing Men's Downhill                      178.211679
Alpine Skiing Men's Giant Slalom                  176.739394
Alpine Skiing Men's Slalom                        176.482143
Alpine Skiing Men's Super G                       179.850575
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.100000
Wrestling Women's Heavyweight, Freestyle          173.444444
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          162.666667
Wrestling Women's Middleweight, Freestyle         164.533333
Name: Height, Length: 726, dtype: float64

Bonus: This dataset contains many athletes more than once (if they competed in multiple events and/or multiple years). You can confirm this by finding .value_counts on the ID column!

In [20]:
df.value_counts()

ID      Age   Height  Team           NOC  Games        Year  Season  City       Sport       Event                                    Medal 
30      26.0  189.0   Netherlands    NED  1996 Summer  1996  Summer  Atlanta    Rowing      Rowing Men's Lightweight Double Sculls   Silver    1
90816   33.0  190.0   Cuba           CUB  2004 Summer  2004  Summer  Athina     Baseball    Baseball Men's Baseball                  Gold      1
92436   31.0  175.0   Spain          ESP  2008 Summer  2008  Summer  Beijing    Sailing     Sailing Mixed Multihull                  Gold      1
92340   26.0  150.0   Belarus        BLR  1996 Summer  1996  Summer  Atlanta    Rowing      Rowing Women's Coxed Eights              Bronze    1
92339   25.0  182.0   Russia         RUS  1996 Summer  1996  Summer  Atlanta    Fencing     Fencing Men's Foil, Team                 Gold      1
                                                                                                                                       

we want to get the average height of all the athletes, but don't want to include any athlete more than once.!
one_per_athlete = df.drop_duplicates(subset = 'Column_Name', keep = 'first')



In [24]:
one_per_athlete = df.drop_duplicates(subset = 'ID', keep = 'first')
one_per_athlete['Height'].mean()

176.31818479042698